src/tree/ResourceGroupTreeItem.ts

```diff
@@ -90,15 +88,13 @@ export class ResourceGroupTreeItem extends GroupTreeItemBase {
         await wizard.execute();
     }
 
-    public compareChildrenImpl(item1: AzExtTreeItem, item2: AzExtTreeItem): number {
-        if ((item1 as AzExtParentTreeItem).loadMoreChildrenImpl && (item2 as AzExtParentTreeItem).loadMoreChildrenImpl) {
+    public compareChildrenImpl(item1: AzExtTreeItem & { collapsibleState: TreeItemCollapsibleState }, item2: AzExtTreeItem & { collapsibleState: TreeItemCollapsibleState }): number {
```

Comment:
This is kind of odd behavior to me.  I know that this is previous behavior, but I'm not really sure what the motivation behind it was.

Reply:
Odd behavior to sort the nodes with children to the top?

Reply:
Yeah, rather than just alphabetical.  I mean, I guess it's pretty visually easy to tell that it's happening, but it doesn't intuitively make sense to me.

Reply:
I'm not sure which one is better, what do you think @fiveisprime?

Tree items with children sorted on top:
![image](https://user-images.githubusercontent.com/12476526/164126461-a1efb06a-9bcd-4a0a-a3b6-f1788258f88c.png)

Purely alphabetical:
![image](https://user-images.githubusercontent.com/12476526/164126795-e958e371-7392-4066-bc4b-971937197c5f.png)

Reply:
My vote would be alphabetical. There is definitely precedent for have-children-first approach (file explorers typically show all folders in alphabetical order, then all files in alphabetical order), but at a high level I don't think of these as "folders" just because they have children. In my mind they are still resources, same as the ones that _don't_ have children.

Reply:
Yeah, I think the way that Brandon explained it is right on the money.  That's exactly why it felt off to me.

Reply:
Sounds like a plan! :shipit:

Reply:
Agreed on alpha for now. I'm interested in experimenting with this further though for sorting by type so that the most commonly used services show up first. I think we can make it more helpful and consistent so that, for example, I always see function apps first and I can expect that.